website/docs/constructs/cookbook/construct-design.md

```diff
@@ -0,0 +1,86 @@
+  
+* Extend `Construct` vs L2 Resource: 
+    * If the construct you want to create contains only one resource and it’s a specific implementation/configuration, then you should go with extending the base resource. An example is the [`AnalyticsBucket`](https://github.com/awslabs/aws-data-solutions-framework/blob/main/framework/src/storage/lib/analytics-bucket.ts#L33).
+    * If the resource contains more than one resource, it should extend `Construct`. It allows to properly scope resources created within the construct (using `this` parameter as the scope) and avoid collisions in resource naming.
+
+
+* Expose all the resources by your construct so you provide a way to end-users to [customize](../customize.mdx) them. Sometimes, it's more user friendly to expose a resource directly to avoid long resources chaining. For example, in the `DataLakeStorage` construct that contains a KMS `Key` attached to the `AnalyticsBucket` part of the `DataLakeStorage`, we re-expose the key so you can access it directly via `dataLakeStorage.key` instead of `dataLakeStorage.bronze.key`.
+
+* If a Construct is creating a role, provide a props to get the role as a parameter. Some AWS customers create roles within a specific process outside of any infrastructure as code.
+
+* Tag resources created with, you can use an aspect:
+    * adsf: version
+    * adsf-layer: layer
+    * adsf-construct: construct
+
+* Bucket encryption: if no business data is store in the bucket, you can use KMS_MANAGED which uses a default KMS key for the account. If the bucket will contain business data, use KMS and create a key.
+
+* Use AWS CDK interfaces (whenever it exists) as input/output of your construct instead of the AWS CDK resource. For example, use `IBucket` instead of `Bucket` as the property or public member of your construct. 
+
+* Avoid using generic `string` parameters for passing values that are  because there is no check during synthetize time.  
+
+* Extend the `TrackedConstruct` instead of the base CDK `Construct` to measure the number of deployments. See the [dedicated documentation](./tracked-construct.md). 
+
+## Coding style
+
+Follow this coding style when submitting code to AWS DSF constructs:
+
+* Use 2 spaces indentation and 120 characters wide
+* ATX style headings in markdown (e.g. ## H2 heading)
+* Import libraries directly 
+    * Use `import { Construct } from '@aws-cdk/core'`
+    * Instead of `import '@aws-cdk/core' as cdk`
+        
+* Do not use Typescript code that is falling into these JSII limitations because JSII won’t be able to translate Typescript code in other languages https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/
```

Comment:
```suggestion
* Do not use Typescript code that is falling into these JSII limitations because JSII won’t be able to translate Typescript code in other languages. See [documentation](https://aws.github.io/jsii/user-guides/lib-author/typescript-restrictions/)
```